### PR TITLE
Add check_attrs & clean-up build.yaml

### PR DESCRIPTION
**Pull request by veblush**
- Add check_attrs.py to verify all attributes of targets are valid in `build.yaml`.
- Remove unnecessary attributes from `build.yaml`

NOTE: This should be merged after #19943 is merged

### CODE REVIEW

File: tools/buildgen/plugins/check_attrs.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+
+def check_attributes(entity, type, errors):
+  attributes = VALID_ATTRIBUTE_KEYS_MAP[type]
+  name = entity.get('name', anything())
+  for key, value in entity.items():
+    if key == 'name':
+      continue
+    validator = attributes.get(key, None)
```
Comment (gnossen) [inline tools/buildgen/plugins/check_attrs.py]: Nit: `dict.get` returns `None` by default, so the second argument is superfluous.

File: tools/buildgen/plugins/check_attrs.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        'timeout_seconds:': anything(),
+        'timeout_seconds': anything(),
+        'uses_polling': anything(),
+        'vs_proj_dir': anything(),
+        'zlib': one_of((True, )),
+    },
+}
+
+
+def check_attributes(entity, type, errors):
```
Comment (gnossen) [inline tools/buildgen/plugins/check_attrs.py]: Nit: `type` is a builtin function in Python. It's not a reserved word, so the code will function, but the function becomes inaccessible for the scope of this function and it trips up syntax highlighters. Perhaps switch to `kind`?

File: tools/buildgen/plugins/check_attrs.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  attributes = VALID_ATTRIBUTE_KEYS_MAP[type]
+  name = entity.get('name', anything())
+  for key, value in entity.items():
+    if key == 'name':
+      continue
+    validator = attributes.get(key, None)
+    if validator:
+      if not validator(value):
+        errors.append("{0}({1}) has an invalid value '{3}' for '{2}'".format(
```
Comment (gnossen) [inline tools/buildgen/plugins/check_attrs.py]: Nit: If you gave the validator a bit more smarts, the error message could be more intelligent, e.g. `'language' attribute must be one of ('linux', 'mac', 'posix', 'windows'), but found 'templeos'`.

File: tools/buildgen/plugins/check_attrs.py (hunk 1 above)
Comment (veblush) [reply tools/buildgen/plugins/check_attrs.py]: That's right!

File: tools/buildgen/plugins/check_attrs.py (hunk 2 above)
Comment (veblush) [reply tools/buildgen/plugins/check_attrs.py]: Oh yeah, I forgot. Done!

File: tools/buildgen/plugins/check_attrs.py (hunk 3 above)
Comment (veblush) [reply tools/buildgen/plugins/check_attrs.py]: Good idea! Done.

**Comment by nicolasnoble** [comment]
Looks like the python script needs formatting.

File: tools/buildgen/plugins/check_attrs.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+  errors = []
+  for filegroup in dictionary.get('filegroups', {}):
+    check_attributes(filegroup, 'filegroup', errors)
+  for lib in dictionary.get('libs', {}):
+    check_attributes(lib, 'lib', errors)
+  for target in dictionary.get('targets', {}):
+    check_attributes(target, 'target', errors)
+  #if errors:
+  #  raise Exception('\n'.join(errors))
```
Comment (nicolasnoble) [inline tools/buildgen/plugins/check_attrs.py]: Why is this still commented out?

Comment (veblush) [reply tools/buildgen/plugins/check_attrs.py]: oh my bad. it should be uncommented. thanks!

**Comment by veblush** [comment]
I reformatted the python source again using yapf. (I did it with cider before and it seems that both are different?) Thanks, Nicolas!

**Pull request merged**